src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -1,49 +1,87 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
 import {createDrawerNavigator} from '@react-navigation/drawer';
+import {withOnyx} from 'react-native-onyx';
 
 import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import {getLastAccessedReport} from '../../reportUtils';
 import styles, {
     getNavigationDrawerType,
     getNavigationDrawerStyle,
 } from '../../../styles/styles';
+import ONYXKEYS from '../../../ONYXKEYS';
+import compose from '../../compose';
+import SCREENS from '../../../SCREENS';
 
 // Screens
 import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import ReportScreen from '../../../pages/home/ReportScreen';
 
 const propTypes = {
+    // Available reports that would be displayed in this navigator
+    reports: PropTypes.objectOf(PropTypes.shape({
+        reportID: PropTypes.number,
+    })),
+
     ...windowDimensionsPropTypes,
 };
 
+const defaultProps = {
+    reports: {},
+};
+
 const Drawer = createDrawerNavigator();
 
-const MainDrawerNavigator = props => (
-    <Drawer.Navigator
-        openByDefault
-        drawerType={getNavigationDrawerType(props.isSmallScreenWidth)}
-        drawerStyle={getNavigationDrawerStyle(
-            props.windowWidth,
-            props.isSmallScreenWidth,
-        )}
-        sceneContainerStyle={styles.navigationSceneContainer}
-        edgeWidth={500}
-        drawerContent={() => <SidebarScreen />}
-    >
-        <Drawer.Screen
-            name="Report"
-            component={ReportScreen}
-
-            // Providing an empty string here will ensure that the ReportScreen does not show as '/r/undefined'
-            // eslint-disable-next-line react/jsx-props-no-multi-spaces
-            initialParams={{reportID: ''}}
-            options={{
+// Decorated to always returning the result of the first call - keeps Screen initialParams from changing
+const getInitialReport = _.once(getLastAccessedReport);
```

Comment:
`_.once` is ideal for initialization scenarios where the value need to be computed just once and always returned since

Reply:
Ok, so we are now waiting for the `reports` to exist instead of a `currentlyViewedReportID`. Makes sense.